html/inc/token.inc

```diff
@@ -0,0 +1,45 @@
+<?php
+// This file is part of BOINC.
+// https://boinc.berkeley.edu
+// Copyright (C) 2018 University of California
+//
+// BOINC is free software; you can redistribute it and/or modify it
+// under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// BOINC is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+// See the GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with BOINC.  If not, see <https://www.gnu.org/licenses/>.
+
+require_once("../inc/boinc_db.inc");
+require_once("../inc/util.inc");
+
+// Constants for valid token types
+define("TOKEN_TYPE_DELETE_ACCOUNT", "D");
+
+// Constants for token durations
+define("TOKEN_DURATION_ONE_DAY", "86400");
+
+function create_confirm_delete_account_token($user) {
```

Comment:
Can this function and the is_valid_delete_account_token be generalized.  I think having them generalized will help decrease the amount of code as well as the maintenance of it going forward.  Having functions that call say create_token($userid, $type, $expire_time = null).  This way anywhere that needs to use a new token can easily call it.  At the moment we have 2 types needed, one for delete account and email change, but this can be used for many more types going forward.  

is_valid_token($userid, $token, $type) is for the other generalized function